### PR TITLE
Update asix-ax88179 with macOS 11.0 driver

### LINKED ISSUE #2226: ASIX AX88179 Big Sur drivers
### 

- [X] I understand that [if I ignore these instructions, my issue may be closed without review](https://github.com/Homebrew/homebrew-cask/blob/master/doc/faq/closing_issues_without_review.md).
- [ ] I have retried my command with `--force`.
- [ ] I ran `brew update-reset && brew update` and retried my command.
- [ ] I ran `brew doctor`, fixed as many issues as possible and retried my command.
- [X] I have checked the instructions for [reporting bugs](https://github.com/Homebrew/homebrew-cask#reporting-bugs).
- [X] I made doubly sure this is not a [checksum does not match](https://github.com/Homebrew/homebrew-cask/blob/master/doc/reporting_bugs/checksum_does_not_match_error.md) error.

### Description of issue

The current AX88179 formula downloads version 2.19.0, which is for macOS 10.15 and below. For Big Sur, there is another driver (possibly to fix the kext driver problem). The latest release for Big Sur is version 1.1.0d2, released on May 11, 2021.

Here is a link to the new download: https://www.asix.com.tw/en/support/download/file/1113

I'm not sure how to edit the formula so that it downloads this new driver for Big Sur only, but the current formula downloads the incorrect driver for the macOS version.

### Command that failed

brew install asix-ax88179

### PR DESCRIPTION
**Pull request by ericswpark**
Closes #2226

**Important:** *Do not tick a checkbox if you haven’t performed its action.* Honesty is indispensable for a smooth review process.

After making all changes to a cask, verify:

- [x] The submission is for [a stable version](https://docs.brew.sh/Acceptable-Casks#stable-versions) or [documented exception](https://docs.brew.sh/Acceptable-Casks#but-there-is-no-stable-version).
- [x] `brew audit --cask {{cask_file}}` is error-free.
- [x] `brew style --fix {{cask_file}}` reports no offenses.

### CODE REVIEW

File: Casks/asix-ax88179.rb [hunk 1]
```diff
@@ -1,5 +1,9 @@
 cask "asix-ax88179" do
-  version "2.19.0,1109"
+  if MacOS.version <= :catalina
+    version "2.19.0,1109"
+  else
+    version "1.2.0,1126"
+  end
   sha256 :no_check
```
Comment (vitorgalvao) [inline Casks/asix-ax88179.rb:7]: This should probably be set.

File: Casks/asix-ax88179.rb [hunk 2]
```diff
@@ -1,5 +1,9 @@
 cask "asix-ax88179" do
-  version "2.19.0,1109"
+  if MacOS.version <= :catalina
```
Comment (vitorgalvao) [inline Casks/asix-ax88179.rb:2]: Instead of using conditionals multiple times, use them *once* and set every specific thing in there.

**Review by vitorgalvao — CHANGES_REQUESTED** [review]

**Comment by ericswpark** [comment]
@vitorgalvao pushed another commit with your suggested changes, could you please take a look? Thanks.

File: Casks/asix-ax88179.rb [hunk 3]
```diff
@@ -1,30 +1,53 @@
...
+      end
     end
-  end
+  else
+    version "1.2.0,1126"
+    sha256 "be21c2ad48572af587502d1286929ed0f7a5b7195d8e249bfbac0b4a8a867244"
 
-  container nested: "AX88179_178A_macOS_10.9_to_10.15_Driver_Installer_v#{version.before_comma}/AX88179_178A_v#{version.before_comma}.dmg"
+    container nested: "ASIX_USB_Device_Installer_macOS_11.0_above_Driver_v#{version.before_comma}/ASIX_USB_Device_Installer_v#{version.before_comma}.dmg"
+    installer manual: "ASIX_USB_Device_Installer_v#{version.before_comma}.pkg"
```
[the remark this answers is not in the document]
Comment (ericswpark) [reply Casks/asix-ax88179.rb:25]: Not sure, it was `installer manual:` originally.

[the remark this answers is not in the document]
Comment (vitorgalvao) [reply Casks/asix-ax88179.rb:25]: Let’s change to PKG. Whatever reason we hadn’t like that may have been fixed by now.

**Review by vitorgalvao — DISMISSED** [review]

**Comment by ericswpark** [comment]
@vitorgalvao it seems like the SHA256 hash doesn't stay constant:
https://github.com/Homebrew/homebrew-cask-drivers/pull/2231/checks?check_run_id=2642746389

**Comment by vitorgalvao** [comment]
> it seems like the SHA256 hash doesn't stay constant:

Probably why it had the `:no_check` in the first place. You can revert that.

**Comment by ericswpark** [comment]
> Probably why it had the `:no_check` in the first place. You can revert that.

Done, just pushed the revert :)

**Review by vitorgalvao — APPROVED** [review]

**Pull request merged**